### PR TITLE
Add unit test to confirm ordering of Codable diags

### PR DESCRIPTION
**Pull request by itaiferber**
**What's in this pull request?**
Adds a FileCheck test to confirm the behavior of #10253. Also catches and fixes a silly diagnostic typo.

### CODE REVIEW

**Comment by itaiferber** [comment]
@swift-ci Please test

**Comment by itaiferber** [comment]
@jrose-apple @DougGregor If this looks reasonable, I'll cherry-pick this into #10334

File: test/decl/protocol/special/coding/class_codable_failure_diagnostics.swift [hunk 1]
```diff
@@ -0,0 +1,140 @@
+// We expect target-swift-frontend to return a status code of 1 here (since it
+// will be outputting diagnostics). That will fail the test, so we have to ||
+// true to set the status code to 0. This necessitates passing the input to
+// FileCheck through an intermediate file.
```
Comment (jrose-apple) [inline test/decl/protocol/special/coding/class_codable_failure_diagnostics.swift]: We have a utility called `not` that avoids this. :-)

```
not %target-swift-frontend … 2>&1 | %FileCheck %s
```

**Review by jrose-apple — COMMENTED** [review]
Seems reasonable to me! Thanks, Itai.

File: test/decl/protocol/special/coding/class_codable_failure_diagnostics.swift (hunk 1 above)
Comment (itaiferber) [inline test/decl/protocol/special/coding/class_codable_failure_diagnostics.swift]: Oh, great! Searched through a bunch of existing tests to look for how this is normally done but I guess I'd missed that. Let me update.

**Comment by itaiferber** [comment]
@swift-ci Please test

**Pull request merged**